TASK-14: Event Card

cristian-aldea opened the pull request:
This PR addressed #62 

You can see the event card by going to `http://localhost:4200/test-event-card`

Notes:
* The join and share buttons are clickable, but they don't do anything for now.
* The number of the date isn't bold since I felt it didn't look right, we can discuss whether the mockup or my implementation needs to be updated.
* There are a few placeholders in this components for now, since much of the implementation it relies on (users joining events, friends, share, etc) isn't implemented yet.
* created new `formatDate` function, which is currently used in event card and event header.

cristian-aldea committed:
Start implementation of event card #62

cristian-aldea committed:
refactor: move formatDate function and change declaration #62

cristian-aldea committed:
Adjust styling and add TODOs #62

cristian-aldea committed:
Implement and update tests #62

Also, fixed tests for event header

cristian-aldea committed:
Fix linting issues #62

r-saba commented on apps/mull-ui/src/app/app.tsx:
```diff
@@ -41,6 +41,15 @@ export const App = () => {
+        {/* Temporary, to be removed */}
+        <Route exact path={'/test-event-card'}>
+          <div className="page-container">
+            <div style={{ padding: '1rem' }}>
```
This padding should be part of `page-container` so all our pages have uniform padding and less repetition in our code.

cristian-aldea replied:
This might work for something like the create event page, but it wouldn't work for the event page, for example. I do agree that styling like this should be repeated as little as possible but idk how

r-saba replied:
We could do what bootstrap does. They have a generic `container` class with default padding. Then you can use `container fluid` if you'd like the container to take up the full screen.

cristian-aldea replied:
Sure, I can do that real quick

r-saba commented on apps/mull-ui/src/app/components/event-card/event-card.tsx:
```diff
@@ -0,0 +1,58 @@
+
+import './event-card.scss';
+
+/* eslint-disable-next-line */
```
We can remove this

r-saba commented on apps/mull-ui/src/app/components/event-card/event-card.tsx:
```diff
@@ -0,0 +1,58 @@
+        <div className="event-card-text">
+          <div className="event-card-title">{event.title}</div>
+          {/* TODO: Replace placeholder distance*/}
+          <div className="event-card-location">14km • {event.location.point}</div>
```
Shouldn't the distance also be dynamic?

cristian-aldea replied:
it is , but currently I have no way of calculating it. I left it a placeholder for now

cristian-aldea replied:
I'll make distance a variable which is interpolated in. And then leave a comment mentionning that it should be implemented.

r-saba replied:
We could do what bootstrap does. They have a generic `container` class with default padding. Then you can use `container fluid` if you'd like the container to take up the full screen.

r-saba replied:
True, it's fine like this for now.

r-saba commented on apps/mull-ui/src/app/components/event-card/event-card.tsx:
```diff
@@ -0,0 +1,58 @@
+          <div className="event-card-location">14km • {event.location.point}</div>
+        </div>
+        {/* TODO: Replace placeholder */}
+        <div className="event-card-friends">friends</div>
```
How come the styling for the friends part of the card is not implemented?

cristian-aldea replied:
I feel like that would be a separate component, and it's used in a few places in the application. I could create a component for it, but the images would all be placeholders

r-saba replied:
I only saw it in event-card if it's in other places it could be a separate task. If not we'd want it to be done in this

cristian-aldea replied:
The event page is also meant to have it, currently, the icon that's there was more meant as a placeholder
![image](https://user-images.githubusercontent.com/23544999/100489965-ba873800-30e5-11eb-85ec-883a625318c3.png)

r-saba replied:
oh okay yeah let's have it as a separate task then, I will make it now

cristian-aldea replied:
thanks ragith

r-saba reviewed (commented):
No blockers just some questions I would like answered

cristian-aldea committed:
Address pr questions #62

bit172 commented on apps/mull-ui/src/app/components/event-page-header/event-page-header.tsx:
```diff
@@ -9,26 +9,16 @@ import './event-page-header.scss';
+  const { day: endDay, month: endMonth, time: endTime } = formatDate(event.endDate);
   return (
     <div className="event-page-header">
       <MullBackButton>{prevPage}</MullBackButton>
```
This is not related to your ticket but I just noticed that pressing the back button reloads the website when going back instead of going to the previous page like a single page application. Would be worth creating a bug ticket for it.

cristian-aldea replied:
Good point, will create one

bit172 reviewed (dismissed):
LGTM! It looks really nice! Good work! 🚀

- [x] Documentation present for relevant new components and functions
- [x] Unit and UI tests are implemented for every component and function
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2000%3A343283)

cristian-aldea committed:
refactor: moved common padding to page-container

Also, added additional classes that can be applied to change the padding/margin of the page-container:
- no-navigation: don't put margin to compensate for the presence of nav elements.
- full-width: no horizontal padding
- no-padding: no padding

cristian-aldea committed:
refactor: Add page-container styling to event page

TheGreatMarkus commented:
@r-saba I addressed your comment about moving common padding to page-container. This affected the styling of all pages which use page-container:
- create event
- login
- register
- test event page
- test event card

So please quickly check these pages to make sure I didn't break them.

@bit172 Sorry for dismissing your review, I created a bug related to what you brought up.

RGPosadas reviewed (dismissed):
Checked out the rest of our pages (login, register, create-event, test-event-page, test-event-card) and they are not broken.

I also like the subtle changes you made compared to the mockup, I find that the event card looks a bit cleaner now. So good job overall! Looks good to me 🚀 

**Acceptance Crtieria**
- [x] Documentation present for relevant new components and functions
- [x]  Unit and UI tests are implemented for every component and function
- [x] Must match Mockup

r-saba reviewed (changes requested):
Noticed there's barely a margin between the `event-card-text` and `event-card-friends` it's noticeable on smaller screen sizes when the text wraps. 
![image](https://user-images.githubusercontent.com/25574985/100524423-11dce500-3186-11eb-9088-a4557de59360.png)
Here's what the white-space looks like on the mockup
![image](https://user-images.githubusercontent.com/25574985/100524455-59637100-3186-11eb-80d1-ffa7e7bc683c.png)

TheGreatMarkus commented:
> Here's what the white-space looks like on the mockup

I agree that adding some more padding can work. Initially, I intentionally made the margin small since wrapping would make it appear much larger due to the added whitespace.

I'm not going to add as much margin as you see on the mockup, since that's obviously due to the mockup card being much longer than the implementation in the example you gave.

And also the mockup appears a bit off since there's very little space between the friend's icon and the share button. That might need to be fixed.

cristian-aldea committed:
Add more margin between friends and other elements #62

cristian-aldea committed:
update snapshots #62

r-saba reviewed (approved):
Thanks for work, it looks really good! LGTM

**Acceptance Criteria**
- [X] Documentation present for relevant new components and functions
- [X] Unit and UI tests are implemented for every component and function
- [X] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2000%3A343283)
